web/src/LogPane.tsx

```diff
@@ -119,11 +183,33 @@ class LogPane extends Component<LogPaneProps, LogPaneState> {
 
     let logLines: Array<React.ReactElement> = []
     let lines = log.split("\n")
-    logLines = lines.map(
-      (line: string, i: number): React.ReactElement => {
-        return <AnsiLine key={"logLine" + i} line={line} />
+
+    let sawBeginning = false
+    let sawEnd = false
+    let highlight = this.props.highlight
+    for (let i = 0; i < lines.length; i++) {
+      const l = lines[i]
+      const key = "logLine" + i
+      let classes = [key]
```

Comment:
it seems deeply weird for each log line to have a custom css class

can you create a single css class ("logLine") and use a data attribute for the key?